coderd/rbac/USAGE.md

```diff
@@ -0,0 +1,344 @@
+# Using RBAC
+
+# Overview
+
+> _NOTE: you should probably read [`README.md`](README.md) beforehand, but it's
+> not essential._
+
+## Basic structure
+
+RBAC is made up of nouns (the objects which are protected by RBAC rules) and
+verbs (actions which can be performed on nouns).<br> For example, a
+**workspace** (noun) can be **created** (verb), provided the requester has
+appropriate permissions.
+
+## Roles
+
+We have a number of roles (some of which have legacy connotations back to v1).
+
+These can be found in `coderd/rbac/roles.go`.
+
+| Role                 | Description                                                         | Example resources (non-exhaustive)           |
+| -------------------- | ------------------------------------------------------------------- | -------------------------------------------- |
+| **owner**            | Super-user, first user in Coder installation, has all\* permissions | all\*                                        |
+| **member**           | A regular user                                                      | workspaces, own details, provisioner daemons |
+| **auditor**          | Viewer of audit log events, read-only access to a few resources     | audit logs, templates, users, groups         |
+| **templateAdmin**    | Administrator of templates, read-only access to a few resources     | templates, workspaces, users, groups         |
+| **userAdmin**        | Administrator of users                                              | users, groups, role assignments              |
+| **orgAdmin**         | Like **owner**, but scoped to a single organization                 | _(org-level equivalent)_                     |
+| **orgMember**        | Like **member**, but scoped to a single organization                | _(org-level equivalent)_                     |
+| **orgAuditor**       | Like **auditor**, but scoped to a single organization               | _(org-level equivalent)_                     |
+| **orgUserAdmin**     | Like **userAdmin**, but scoped to a single organization             | _(org-level equivalent)_                     |
+| **orgTemplateAdmin** | Like **templateAdmin**, but scoped to a single organization         | _(org-level equivalent)_                     |
+
+_\* except some, which are not important to this overview_
```

Comment:
The `Example resources` does not indicate what actions the role can do on said resources. Saying `member `has `provisioner` resource is correct, but it is only `read` access such that they can provision their workspace. This table does not indicate which actions are granted, so it might appear that `templates == full CRUD` when it does not.

It might be confusing at this point in the doc to list the nuanced permissions that are required for the role to work, but probably not intuitive in the outright. An example is user -> provisioner. This is required because a user needs to read the provisioner to know if their workspace can be provisioned, but it does not grant CRUD over said provisioners.

It should be clarified somehow that these resources have various limitations based on the roles, or if we just listed the more unique aspects of each role in relation to each other. 

```suggestion
| Role                 | Description                                                         | Example resources (non-exhaustive)           |
| -------------------- | ------------------------------------------------------------------- | -------------------------------------------- |
| **owner**            | Super-user, first user in Coder installation, has all\* permissions | all\*                                        |
| **member**           | A regular user                                                      | workspaces, own details, provisioner daemons |
| **auditor**          | Viewer of audit log events, read-only access to a few resources     | audit logs, templates, users, groups         |
| **templateAdmin**    | Administrator of templates, read-only access to a few resources     | templates, workspaces, users, groups         |
| **userAdmin**        | Administrator of users                                              | users, groups, role assignments              |
| **orgAdmin**         | Like **owner**, but scoped to a single organization                 | _(org-level equivalent)_                     |
| **orgMember**        | Like **member**, but scoped to a single organization                | _(org-level equivalent)_                     |
| **orgAuditor**       | Like **auditor**, but scoped to a single organization               | _(org-level equivalent)_                     |
| **orgUserAdmin**     | Like **userAdmin**, but scoped to a single organization             | _(org-level equivalent)_                     |
| **orgTemplateAdmin** | Like **templateAdmin**, but scoped to a single organization         | _(org-level equivalent)_                     |

**Note an example resource indicates the role has at least 1 permission related to the resource. Not that the role has complete CRUD access to the resource.**

_\* except some, which are not important to this overview_
```